src/org/zaproxy/zap/extension/httpsinfo/HttpsInfoDialog.java

```diff
@@ -0,0 +1,339 @@
+	}
+	
+	private JTextArea getGeneralTextArea() {
+		return general;
+	}
+	
+	private JTextArea getCipherSuitesTextArea() {
+		return cipherSuites;
+	}
+
+	public void init() {
+		checkProxyChainEnabled();
+		
+		initGeneral();
+		initCipherSuites();
+	}
+	
+	private void initGeneral() {
+		general = new JTextArea();
+		general.setEditable(false);
+		general.setRows(12);
+		general.setColumns(70);
+		general.setLineWrap(true);
+	}
+	
+	private void initCipherSuites() {
+		cipherSuites = new JTextArea();
+		cipherSuites.setEditable(false);
+		cipherSuites.setRows(15);
+		cipherSuites.setColumns(70);
+	}
+	
+	private void setTarget(HttpMessage msg) {
+		try {
+			this.target = new URL(msg.getRequestHeader().getURI().toString());
+		} catch (MalformedURLException e) {
+			LOGGER.warn("An exception occurred while attempting to set the target", e);
+		}
+	}
+	
+	private URL getTarget() {
+		if (target == null) {
+			LOGGER.warn("Somehow the target was not set, when we tried to use it.");
+			return null;
+		}
+		return target;
+	}
+	
+	private void setBaseMessage(HttpMessage msg) {
+		this.baseMessage = msg;
+	}
+	
+	private HttpMessage getBaseMessage() {
+		return baseMessage;
+	}
+	
+	private void initSession(URL target) throws DVException {
+		try {
+			this.session = DVFactory.initializeSession(target);
+		} catch (DVException e) {
+			throw new DVException("An exception occurred while initializing the DV session. " + e.getMessage(),
+					e.getCause());
+		}
+	}
+	
+	private IDVSession getSession() {
+		return session;
+	}
+	
+	private void setDvEng(IDVSession session) {
+		this.dvEng = DVFactory.getDVEng(session);
+	}
+	
+	private IDVEng getDvEng() {
+		return dvEng;
+	}
+	
+	private void doThreadedTasks() {
+		Thread httpsInfoThread = new Thread("httpsinfo") {
+			public void run() {
+				try {
+					initSession(getTarget());
+				} catch (DVException e) {
+					String warnMsg = Constant.messages.getString("httpsinfo.init.warning", getTarget().toString(),
+							e.getCause()); 
+					LOGGER.warn(warnMsg);
+					getGeneralTextArea().setText(warnMsg);
+					return;
+				}
+				setDvEng(getSession());
+				showGeneral();
+				showCipherSuites();
+			}
+		};
+		httpsInfoThread.start();
+	}
+
+	private void showGeneral() {
+		StringBuilder content = new StringBuilder(
+				Constant.messages.getString("httpsinfo.general.server.leadin", target.getHost()));
+		try {
+			if (getDvEng().getCertificate() == null) {
+				content.append(Constant.messages.getString("httpsinfo.general.cert.notfound"));
+			} else {
+				content.append(Constant.messages.getString("httpsinfo.general.cert.heading"));
+				content.append("  ").append(getCleanCertStringRepresentation(getDvEng().getCertificate())).append(NEWLINE);
+			}
+		} catch (DVException e) {
+			String generalException = Constant.messages.getString("httpsinfo.general.exception",
+					e.getMessage());
+			LOGGER.warn(generalException, e);
+			getGeneralTextArea().setText(generalException);
```

Comment:
Should return after this statement (or the following two statements moved to try block).